### DISCUSSION TITLE
Error writing unto pre-built container image

### DISCUSSION THREAD

**Discussion by richardanarfi** [princeton-vl/CoqGym#68] [Q&A] [answered]
Hello,

Is the pre-built container image read-only?

I wanted to test this:
python -u evaluate.py auto auto-TEST --file ../data/StructTact/Assoc.json --proof "get_set_same"

I get this error at the end:
OSError: [Errno 30] Read-only file system: 'evaluation/auto-TEST'

**Comment by yangky11** [comment]
Sorry I've been swamped recently and can probably take a look this weekend.

**Comment by yangky11** [comment, marked as answer]
This turns out to be a simply Singularity issue. The container is read-only by default. But you can use the ` --writable` option when running the `singularity shell` command to make it writable. Please refer to Singularity's documentation for details.
